server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2142,12 +2142,27 @@ public boolean isReadOnly() {
     }
 
     /**
-     * Writing a new index generation is a three step process.
-     * First, the {@link RepositoryMetadata} entry for this repository is set into a pending state by incrementing its
-     * pending generation {@code P} while its safe generation {@code N} remains unchanged.
-     * Second, the updated {@link RepositoryData} is written to generation {@code P + 1}.
-     * Lastly, the {@link RepositoryMetadata} entry for this repository is updated to the new generation {@code P + 1} and thus
-     * pending and safe generation are set to the same value marking the end of the update of the repository data.
+     * Writing a new index generation (root) blob is a three-step process:
+     * <ol>
+     * <li>
+     * The {@link RepositoryMetadata} entry for this repository is set into a pending state by increasing its pending generation {@code P}
+     * while its safe generation {@code N < P} remains unchanged.
```

Comment:
Can we somehow make it more explicit that N and P begin as equal numbers and throughout their lifecycles, N will always be less than or equal to P?

Reply:
I added some more details on this in 8c798f9ad53005996b5e1cdf19345745d07c0b1d. It's not true that N and P always start out equal, although that's typically the case.